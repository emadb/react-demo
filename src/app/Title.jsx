export default function Title(props) {
  return <h3 className="title">{props.text}</h3>
}

Title.defaultProps = { text: "Title here"}