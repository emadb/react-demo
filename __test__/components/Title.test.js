import React from 'react'
import renderer from 'react-test-renderer'
import Title from '../../src/app/Title'

describe('<Title />', () => {
  
  test('just render with default props', () => {
    const component = renderer.create(<Title />)
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('just render with a custom text', () => {
    const component = renderer.create(<Title text="Hello world" />)
    expect(component.toJSON()).toMatchSnapshot()
  })

})