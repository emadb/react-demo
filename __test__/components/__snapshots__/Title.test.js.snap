// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> just render with a custom text 1`] = `
<h3
  className="title"
>
  Hello world
</h3>
`;

exports[`<Title /> just render with default props 1`] = `
<h3
  className="title"
>
  Title here
</h3>
`;
